pandas/core/internals/blocks.py

```diff
@@ -612,9 +612,15 @@ def replace(
 
         else:
             # split so that we only upcast where necessary
-            return self.split_and_operate(
-                type(self).replace, to_replace, value, inplace=True
-            )
+            blocks = []
+            nbs = self._split()
```

Comment:
could write this entire section as a list-comprehension, but maybe not as readable

Reply:
good idea, i like the list comprehension better

Reply:
Went back to a non-list comprehension implementation as it requires unpacking.